src/main/java/pl/edu/agh/ki/io/api/providers/AchievementsProvider.java

```diff
@@ -0,0 +1,146 @@
+        }
+
+        for (Map.Entry<Integer, Integer> daysCompletions : Map.of(14, 2, 30, 5, 7, 2).entrySet()) {
+            Achievement achievement = new Achievement();
+            achievement.setAchievementTitle(numberTimeConstrainedPeakCompletionTemplate.formatted(
+                    daysCompletions.getValue(),
+                    daysCompletions.getValue() != 2 ? "ów" : "y",
+                    daysCompletions.getKey()
+            ));
+            achievement.setToComplete(daysCompletions.getValue());
+
+            int completedInTime = 0;
+            for (PeakWithCompletion peakCompletion : peakWithCompletions){
+                if (peakCompletion.getCreateDate() == null) break;
+
+                long lowerLimit = peakCompletion.getCreateDate().getTime() - TimeUnit.DAYS.toMillis(daysCompletions.getKey());
+                completedInTime = Math.max(completedInTime, peakWithCompletions.stream().map(completion ->
+                        completion.getCreateDate() != null && completion.getCreateDate().getTime() >= lowerLimit ? 1 : 0).reduce(0, Integer::sum));
+            }
+
+            if (completedInTime >= daysCompletions.getValue()) {
+                achievement.setProgress(daysCompletions.getValue());
+                achievement.setCompleted(true);
+            } else {
+                achievement.setProgress(completedInTime);
+                achievement.setCompleted(false);
+            }
+            achievements.add(achievement);
+        }
+
+        return achievements;
+    }
+
+    public List<Achievement> getAchievementsShort(User user) {
+        List<Achievement> achievements = getAchievements(user);
+
+        List<Achievement> completed = achievements.stream().filter(Achievement::isCompleted).limit(5).collect(Collectors.toList());
+        List<Achievement> uncompleted = achievements.stream().filter(achievement -> !achievement.isCompleted()).sorted().limit(5).collect(Collectors.toList());
+
+        List<Achievement> shortAchievements = new LinkedList<>();
+        if(completed.size() < 2){
+            shortAchievements.addAll(completed);
+            shortAchievements.addAll(uncompleted.subList(0, 5 - completed.size()));
+        } else {
+            if (uncompleted.size() < 3) {
+                shortAchievements.addAll(completed.subList(0, 5 - uncompleted.size()));
+                shortAchievements.addAll(uncompleted);
+            } else {
+                shortAchievements.addAll(completed.subList(0 ,2));
+                shortAchievements.addAll(uncompleted.subList(0, 3));
+            }
+        }
+        return shortAchievements;
+    }
+
+    @Data
+    @AllArgsConstructor
+    @NoArgsConstructor
+    public static class Achievement implements Comparable<Achievement>{
+        private String achievementTitle;
+        private boolean completed = false;
+        private int progress;
+        private int toComplete;
+        private Long peakId;
+        private Long timeLeft;
```

Comment:
To jest gdzieś wykorzystywane?

Reply:
oops, mogło by być ale nie jest, chyba i tak. tego nie użyjecie, już usuwam